extensions/target-specific/chromium/blink/blink-helpers/blink-helpers.js

```diff
@@ -551,7 +551,8 @@ Loader.OnLoad(function() {
             description: "Returns (a promise to) a collection of documents for all web frames in the current renderer process."
         },
         GetDocuments: () => {
-            return DbgObject.global(Chromium.RendererProcessSyntheticModuleName, "g_frame_map")
+            var module = Chromium.RendererProcessSyntheticModuleName;
+            return Promise.any([DbgObject.global(module, "g_frame_map"), DbgObject.global(module, "content::(anonymous namespace)::g_frame_map")])

```

Comment:
Is this an issue for all globals? Is there something we can change about DbgObject.global to avoid every caller having to do this?

Reply:
It's an issue for all globals that are defined in a namespace.

Hm, maybe we could have the JS code always pass the value with the namespace and have the server remove the namespace for windbg/visual studio?

Or yes, we could have DbgObject.global always do an automatic .any

Thoughts on those options?

Reply:
I think adding an optional namespace parameter to DbgObject.global is reasonable.

In general, the more we can do in JS the better. However, if there is a perf benefit to performing one symbolic lookup instead of two, then making a server side change may be worth considering.

Just to clarify, does the DIA global symbol lookup (what's used on Windows) not work if you look up a global symbol with its namespace?

Reply:
Yes, doesn't work with namespace in WinDbg:
```0:042> dt content_shell!g_frame_map
*** WARNING: Unable to verify checksum for content_shell.exe
DBGHELP: content_shell - private symbols & lines 
        c:\src\src\out\gn\content_shell.exe.pdb
   +0x000 private_instance_ : 0n140696262961808
   +0x008 private_buf_     : [16]  "???"
0:042> dt content_shell!content::(anonymous namespace)::g_frame_map
Symbol content_shell!content::(anonymous namespace)::g_frame_map not found.
0:042> dt content_shell!content::g_frame_map
Symbol content_shell!content::g_frame_map not found.
```

The advantage of doing it on the server is that it knows for sure whether the namespace is needed. But anyway, I'll add it to DbgObject.global for now.

Reply:
Please take another look -- now on DbgObject.global

Reply:
Now that I've thought about this some more, I think we should keep the namespace parameter on DbgObject.global, but do the rest of the handling server-side.

Instead of trying to look up both symbols and seeing which one works, we should pass the namespace parameter from DbgObject.global, through to LookupGlobalSymbol, then down to the JsDbg server into LookupGlobalSymbol on IDebugger. GdbDebugger and TypeCacheDebugger both implement IDebugger - TypeCacheDebugger can just ignore the namespace parameter and GdbDebugger can use it.

I did something similar to add the typename parameter to DbgObject.global recently (https://github.com/MicrosoftEdge/JsDbg/commit/2743129686132e507a4619a4971631a53c3b581c) - you can follow the same pattern.

I'd also recommend merging master into the GDB branch first to avoid merge conflicts. Let me know if it is already a nightmare - I can help out.

Reply:
OK, pull request for the merge is at https://github.com/MicrosoftEdge/JsDbg/pull/46

Reply:
https://github.com/MicrosoftEdge/JsDbg/pull/49 adds the API; once that's merged I will update this PR to implement it for GDB & use it in blink-helpers.